### PR TITLE
Feature/1263 - Added Session Store

### PR DESCRIPTION
**Pull request by ygit**
# Description

Added new Session Store APIs

Closes

- #1263 

---

## Additional context

* Tested Tile Spotlight feature on Flutter iOS & Web
* Need to test tile spotlight and pin message functionality in the application across platforms

### Pre-launch Checklist

- [ ] The [Documentation] is updated accordingly, or this PR doesn't require it.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I added new tests to check the change I am making, or this PR is test-exempt.
- [x] All existing and new tests are passing.

[Documentation]: https://www.100ms.live/docs
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement

### CODE REVIEW

File: ios/Classes/SwiftHmssdkFlutterPlugin.swift [hunk 1]
```diff
@@ -14,17 +14,24 @@ public class SwiftHmssdkFlutterPlugin: NSObject, FlutterPlugin, HMSUpdateListene
...
+
     var roleChangeRequest: HMSRoleChangeRequest?
 
     internal var hmsSDK: HMSSDK?
 
     private var isStatsActive = false
 
+    internal var sessionStore: HMSSessionStore?
+
+    private var sessionStoreChangeObservers = [HMSSessionStoreKeyChangeListener]()
```
Comment (ygit) [inline ios/Classes/SwiftHmssdkFlutterPlugin.swift:33]: @Decoder07 this `sessionStoreChangeObservers` can be used to set single or multiple observers for different or same keys

Comment (Decoder07) [reply ios/Classes/SwiftHmssdkFlutterPlugin.swift:33]: but how can we identify where to give(in which class) the updates.

File: ios/Classes/Actions/HMSSessionStoreAction.swift [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+        }
+
+        guard let arguments = call.arguments as? [AnyHashable: Any],
+            let key = arguments["key"] as? String
+        else {
+            HMSErrorLogger.returnHMSException(#function,"Key for the object to be set in Session Store is null.","NULL ERROR",result)
+            return
+        }
+
+        guard let data = arguments["data"]
```
Comment (ygit) [inline ios/Classes/Actions/HMSSessionStoreAction.swift:76]: it should be possible to set nil values as well

File: android/src/main/kotlin/live/hms/hmssdk_flutter/HMSErrorLogger.kt [hunk 3]
```diff
@@ -24,6 +24,8 @@ class HMSErrorLogger {
          */
         fun returnArgumentsError(errorMessage: String): Unit? {
             HMSLogger.e("FL_HMSSDK Args Error", errorMessage)
+
+            // TODO: why is this return null required?
```
Comment (ygit) [inline android/src/main/kotlin/live/hms/hmssdk_flutter/HMSErrorLogger.kt:28]: @Decoder07 why is this return null required?

Comment (Decoder07) [reply android/src/main/kotlin/live/hms/hmssdk_flutter/HMSErrorLogger.kt:28]: This is required because a value needs to be returned to the value

**Review by Decoder07 — APPROVED** [review]

**Pull request merged**